test/unit/mdc-select/foundation.test.js

```diff
@@ -164,15 +167,145 @@ test('#handleFocus calls adapter.activateBottomLine()', () => {
   td.verify(mockAdapter.activateBottomLine(), {times: 1});
 });
 
+test('#handleFocus calls adapter.activateBottomLine() if isMenuOpen=true', () => {
+  const {foundation, mockAdapter} = setupTest();
+  td.when(mockAdapter.isMenuOpen()).thenReturn(true);
+  foundation.handleFocus();
+  td.verify(mockAdapter.activateBottomLine(), {times: 1});
+});
+
 test('#handleBlur calls foundation.handleChange()', () => {
   const {foundation} = setupTest();
   foundation.handleChange = td.func();
   foundation.handleBlur();
-  td.verify(foundation.handleChange(), {times: 1});
+  td.verify(foundation.handleChange(false), {times: 1});
 });
 
 test('#handleBlur calls adapter.deactivateBottomLine()', () => {
   const {foundation, mockAdapter} = setupTest();
   foundation.handleBlur();
   td.verify(mockAdapter.deactivateBottomLine(), {times: 1});
 });
+
+test('#handleBlur does not call deactivateBottomLine if isMenuOpen=true', () => {
+  const {foundation, mockAdapter} = setupTest();
+  td.when(mockAdapter.isMenuOpen()).thenReturn(true);
+  foundation.handleBlur();
+  td.verify(mockAdapter.deactivateBottomLine(), {times: 0});
+});
+
+test('#handleClick does nothing if isMenuOpen=true', () => {
+  const {foundation, mockAdapter} = setupTest();
+  td.when(mockAdapter.isMenuOpen()).thenReturn(true);
+  foundation.handleClick(0);
+  td.verify(mockAdapter.setRippleCenter(0), {times: 0});
+});
+
+test('#handleClick sets the ripple center if isMenuOpen=false', () => {
+  const {foundation, mockAdapter} = setupTest();
+  td.when(mockAdapter.isMenuOpen()).thenReturn(false);
+  foundation.handleClick(0);
+  td.verify(mockAdapter.setRippleCenter(0), {times: 1});
+});
+
+test('#handleClick opens the menu if the select is focused and isMenuOpen=false', () => {
+  const {foundation, mockAdapter} = setupTest();
+  td.when(mockAdapter.isMenuOpen()).thenReturn(false);
+  td.when(mockAdapter.hasClass(td.matchers.anything())).thenReturn(true);
```

Comment:
change `td.matchers.anything()` to `cssClasses.FOCUSED`? It's unclear whether there are intentional or unintentional side effects of this matcher.